### ISSUE TITLE
Extend InfinityModel

### ISSUE THREAD

**Issue by snewcomer** [hhff/ember-infinity#222]
- [x] Expose infinityModel constructor to people to extend in their own apps  
- [x] Allow them to pass a class into the infinityModel that should be extended - i.e. as third argument in Route.
- [x] improve documentation (also alluding to #97 as well)
- [ ] public getter for infinityModels.  Something like `route#getInfinityModels()`

```
import InfinityModel from 'ember-infinity/lib/infinity-model';
const CustomInfinityModel = InfinityModel.extend({
  //..snip
})

export default Route.extend({
  model() {
    return this.infinityModel('product, { perPage: 10, startingPage: 1 }, CustomInfinityModel);
  }
})
```

Ref https://github.com/hhff/ember-infinity/pull/220

Will tackle after #220

**Comment by snewcomer** [comment]
More notes
```
// new way

const MyInfinityModelClass = InfinityModel.extend({
  global: service(),
  buildParams() {
    let params = this._super(...arguments);
    params['category_id'] = get(this, 'global.categoryId');
    return params;
  }
});

Route.extend({
  global: service(),
  categoryId: computed('global.categoryId', function() {
    return get(this, 'global.categoryId');
  }),
  model() {
    this.infinityModel('product', {}, MyInfinityModelClass);
  }
});
```

**Comment by snewcomer** [comment]
I'll make sure I get this issue done by the end of next week to add to the 1-0 branch.

**Comment by hhff** [comment]
Thankyou!!

**Comment by snewcomer** [comment]
closing this after #236 was merged.  Still have one checkbox outstanding.  Jotting that down for another issue.
